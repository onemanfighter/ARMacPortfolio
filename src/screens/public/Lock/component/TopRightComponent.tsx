import { BatteryIcon, CheckIcon, KeyboardIcon, WifiIcon } from '@assets';
import {
  Box,
  Button,
  Divider,
  Popover,
  PopoverBody,
  PopoverContent,
  PopoverTrigger,
  Portal,
} from '@chakra-ui/react';
import { useTranslation } from 'react-i18next';

const TopRightComponent = () => {
  const { t } = useTranslation();

  return (
    <Box
      width={'100%'}
      alignItems={'center'}
      position={'fixed'}
      top={'0'}
      right={'0'}
      display={'flex'}
      justifyContent={'flex-end'}
      marginEnd={'10'}
      zIndex={0}
      aria-label="top-right-component"
      gap={2}
    >
      <Popover>
        <PopoverTrigger>
          <Button
            rightIcon={
              <KeyboardIcon color="white" aria-label="keyboard-icon" />
            }
            size={'xs'}
            variant="none"
            color={'white'}
            aria-label="keyboard-icon-button"
          >
            {t('LockScreen.keyboard.type')}
          </Button>
        </PopoverTrigger>
        <Portal>
          <PopoverContent width={'auto'} bg="#0000007f">
            <PopoverBody gap={1}>
              <Button
                size="xs"
                width="100%"
                color="white"
                bg="transparent"
                _hover={{ color: 'black', bg: 'white' }}
                rightIcon={<CheckIcon />}
              >
                {t('LockScreen.keyboard.option.india')}
              </Button>
              <Divider my={1} />
              <Button size={'xs'} variant="none" color={'white'} disabled>
                {t('LockScreen.keyboard.menu.others')}
              </Button>
            </PopoverBody>
          </PopoverContent>
        </Portal>
      </Popover>
      <BatteryIcon
        percentage={67}
        props={{
          color: 'white',
          'aria-label': 'battery-icon',
          width: 16,
          height: 16,
        }}
      />
      <WifiIcon color="white" aria-label="wifi-icon" />
    </Box>
  );
};

export default TopRightComponent;
