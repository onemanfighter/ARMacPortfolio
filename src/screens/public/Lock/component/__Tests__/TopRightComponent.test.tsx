import { render, screen } from '@testing-library/react';
import TopRightComponent from '../TopRightComponent';
import userEvent from '@testing-library/user-event';

describe('TopRightComponent', () => {
  it('should render correctly to match snapshot', () => {
    const { container } = render(<TopRightComponent />);
    expect(container).toMatchSnapshot();
  });

  it('should render the keyboard icon', () => {
    render(<TopRightComponent />);

    expect(screen.getByLabelText('keyboard-icon')).toBeDefined();
  });

  it('should render the battery icon ', () => {
    render(<TopRightComponent />);

    expect(screen.getByLabelText('battery-icon')).toBeDefined();
  });

  it('should render the wifi icon ', () => {
    render(<TopRightComponent />);

    expect(screen.getByLabelText('wifi-icon')).toBeDefined();
  });

  it('should render the keyboard icon button', () => {
    render(<TopRightComponent />);
    userEvent.click(screen.getByLabelText('top-right-component'));

    expect(screen.getByLabelText('keyboard-icon-button')).toBeDefined();
  });

  it('should render the popover on clicking keyboard icon', () => {
    render(<TopRightComponent />);
    userEvent.click(screen.getByLabelText('keyboard-icon'));

    expect(screen.getByText('Other source inputs')).toBeDefined();
  });
});
