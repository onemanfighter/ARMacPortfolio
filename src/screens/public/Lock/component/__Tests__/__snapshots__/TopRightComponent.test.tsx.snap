// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopRightComponent should render correctly to match snapshot 1`] = `
<div>
  <div
    aria-label="top-right-component"
    class="css-sikaw0"
  >
    <button
      aria-controls="popover-content-:r1:"
      aria-expanded="false"
      aria-haspopup="dialog"
      aria-label="keyboard-icon-button"
      class="chakra-button css-117sk0y"
      id="popover-trigger-:r1:"
      type="button"
    >
      ABC - India
      <span
        class="chakra-button__icon css-1hzyiq5"
      >
        <svg
          aria-hidden="true"
          aria-label="keyboard-icon"
          color="white"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M14 5a1 1 0 011 1v5a1 1 0 01-1 1H2a1 1 0 01-1-1V6a1 1 0 011-1h12zM2 4a2 2 0 00-2 2v5a2 2 0 002 2h12a2 2 0 002-2V6a2 2 0 00-2-2H2z"
          />
          <path
            d="M13 10.25a.25.25 0 01.25-.25h.5a.25.25 0 01.25.25v.5a.25.25 0 01-.25.25h-.5a.25.25 0 01-.25-.25v-.5zm0-2a.25.25 0 01.25-.25h.5a.25.25 0 01.25.25v.5a.25.25 0 01-.25.25h-.5a.25.25 0 01-.25-.25v-.5zm-5 0A.25.25 0 018.25 8h.5a.25.25 0 01.25.25v.5a.25.25 0 01-.25.25h-.5A.25.25 0 018 8.75v-.5zm2 0a.25.25 0 01.25-.25h1.5a.25.25 0 01.25.25v.5a.25.25 0 01-.25.25h-1.5a.25.25 0 01-.25-.25v-.5zm1 2a.25.25 0 01.25-.25h.5a.25.25 0 01.25.25v.5a.25.25 0 01-.25.25h-.5a.25.25 0 01-.25-.25v-.5zm-5-2A.25.25 0 016.25 8h.5a.25.25 0 01.25.25v.5a.25.25 0 01-.25.25h-.5A.25.25 0 016 8.75v-.5zm-2 0A.25.25 0 014.25 8h.5a.25.25 0 01.25.25v.5a.25.25 0 01-.25.25h-.5A.25.25 0 014 8.75v-.5zm-2 0A.25.25 0 012.25 8h.5a.25.25 0 01.25.25v.5a.25.25 0 01-.25.25h-.5A.25.25 0 012 8.75v-.5zm11-2a.25.25 0 01.25-.25h.5a.25.25 0 01.25.25v.5a.25.25 0 01-.25.25h-.5a.25.25 0 01-.25-.25v-.5zm-2 0a.25.25 0 01.25-.25h.5a.25.25 0 01.25.25v.5a.25.25 0 01-.25.25h-.5a.25.25 0 01-.25-.25v-.5zm-2 0A.25.25 0 019.25 6h.5a.25.25 0 01.25.25v.5a.25.25 0 01-.25.25h-.5A.25.25 0 019 6.75v-.5zm-2 0A.25.25 0 017.25 6h.5a.25.25 0 01.25.25v.5a.25.25 0 01-.25.25h-.5A.25.25 0 017 6.75v-.5zm-2 0A.25.25 0 015.25 6h.5a.25.25 0 01.25.25v.5a.25.25 0 01-.25.25h-.5A.25.25 0 015 6.75v-.5zm-3 0A.25.25 0 012.25 6h1.5a.25.25 0 01.25.25v.5a.25.25 0 01-.25.25h-1.5A.25.25 0 012 6.75v-.5zm0 4a.25.25 0 01.25-.25h.5a.25.25 0 01.25.25v.5a.25.25 0 01-.25.25h-.5a.25.25 0 01-.25-.25v-.5zm2 0a.25.25 0 01.25-.25h5.5a.25.25 0 01.25.25v.5a.25.25 0 01-.25.25h-5.5a.25.25 0 01-.25-.25v-.5z"
          />
        </svg>
      </span>
    </button>
    <svg
      aria-label="battery-icon"
      color="white"
      fill="currentColor"
      height="16"
      viewBox="0 0 16 16"
      width="16"
    >
      <path
        d="M2 6h6.7v4H2V6z"
      />
      <path
        d="M2 4a2 2 0 00-2 2v4a2 2 0 002 2h10a2 2 0 002-2V6a2 2 0 00-2-2H2zm10 1a1 1 0 011 1v4a1 1 0 01-1 1H2a1 1 0 01-1-1V6a1 1 0 011-1h10zm4 3a1.5 1.5 0 01-1.5 1.5v-3A1.5 1.5 0 0116 8z"
      />
    </svg>
    <svg
      aria-label="wifi-icon"
      color="white"
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M12 21l3.6-4.8c-1-.75-2.25-1.2-3.6-1.2s-2.6.45-3.6 1.2L12 21m0-18C7.95 3 4.21 4.34 1.2 6.6L3 9c2.5-1.88 5.62-3 9-3s6.5 1.12 9 3l1.8-2.4C19.79 4.34 16.05 3 12 3m0 6c-2.7 0-5.19.89-7.2 2.4l1.8 2.4C8.1 12.67 9.97 12 12 12c2.03 0 3.9.67 5.4 1.8l1.8-2.4C17.19 9.89 14.7 9 12 9z"
      />
    </svg>
  </div>
</div>
`;
